packages/teuchos/core/src/Teuchos_RCPNode.cpp

```diff
@@ -405,12 +414,12 @@ void RCPNodeTracer::printActiveRCPNodes(std::ostream &out)
   }
 }
 
-
-// Internal implementation functions
-
-
 void RCPNodeTracer::addNewRCPNode( RCPNode* rcp_node, const std::string &info )
 {
+#ifdef USE_MUTEX_TO_PROTECT_NODE_TRACING
+	rcp_node_list_mutex()->lock();

```

Comment:
Replace this try/catch and direct call of lock() with:

```
std::lock_guard<std::mutex> lock(rcp_node_list_mutex());
```
See:

http://en.cppreference.com/w/cpp/thread/lock_guard